modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -1037,9 +1054,15 @@
 (defn- handle-order-by [{:keys [order-by breakout]} pipeline-ctx]
   (let [breakout-fields (set breakout)
         sort-fields (for [field (remove-parent-fields (map second order-by))
-                          ;; We only care about expressions not added as breakout
+                          ;; We only care about expressions and bucketing not added as breakout
                           :when (and (not (contains? breakout-fields field))
-                                     (= :expression ((.dispatchFn ^clojure.lang.MultiFn ->rvalue) field)))]
+                                     (let [dispatch-value
+                                           ((.dispatchFn ^clojure.lang.MultiFn ->rvalue) field)]
```

Comment:
I know you didn't write this bit but it feels like this should be another multimethod, rather than using undocumented methods of the implementation to get the dispatch function. We could also just extract the dispatch function and reference that normally here.

Reply:
i.e. we can just use `mbql.u/dispatch-by-clause-name-or-class` instead of the `.dispatchFn` call here, or better yet, a top-level `def` to reference it here and in the `defmulti`.

Reply:
I did the first option. I don't think the consistence with the dispatcher function is essential here. I think we need `mbql.u/dispatch-by-clause-name-or-class` and used that.